lib/WebSocketServer.js

```diff
@@ -40,7 +40,7 @@ function WebSocketServer(options, callback) {
     maxPayload: null
   }).merge(options);
 
-  if (!options.isDefinedAndNonNull('port') && !options.isDefinedAndNonNull('server') && !options.value.noServer) {
+  if (!options.value.noServer && (!options.isDefinedAndNonNull('port') || !options.isDefinedAndNonNull('server'))) {
```

Comment:
It doesn't seem correct to me. For example:
```js
const http = require('http');
const ws = require('ws');

const server = http.createServer();
const wss = new ws.Server({ server });

...
```
throws with this change but it shouldn't.